src/main.c

```diff
@@ -154,14 +153,15 @@ main (int    argc,
 
   loop = g_main_loop_new (NULL, FALSE);
 
-  sigint_id = 0;
   if (!opt_no_sigint)
     {
-      sigint_id = g_unix_signal_add_full (G_PRIORITY_DEFAULT,
+      guint sigint_id = g_unix_signal_add_full (G_PRIORITY_DEFAULT,
```

Comment:
@vpodzime The original design appears to disregard any protection to concurrent access to this hash table. From looking at the code and considering what I am seeing with #110 I'm starting to think that having concurrent access to the same hash table is indeed possible, which would be bad as the GHashTable is not thread safe.  I was planning on looking at #110 some more, so when I am looking at that I will see if I can convince myself that this issue does or does not exist (concurrent access to same hash table).